libs/server-sdk/src/bindings/c/builder.cpp

```diff
@@ -236,6 +255,45 @@ LDServerDataSourcePollBuilder_Free(LDServerDataSourcePollBuilder b) {
     delete TO_POLL_BUILDER(b);
 }
 
+LD_EXPORT(LDServerLazyLoadBuilder)
+LDServerLazyLoadBuilder_New() {
+    return FROM_LAZYLOAD_BUILDER(new DataSystemBuilder::LazyLoad());
+}
+
+LD_EXPORT(void)
+LDServerLazyLoadBuilder_Free(LDServerLazyLoadBuilder b) {
+    delete TO_LAZYLOAD_BUILDER(b);
+}
+
+LD_EXPORT(void)
```

Comment:
Wasn't entirely sure what to do here. Normally we'd pass in a struct with some function pointers, to allow the C users to truly implement the interface - but that would require a ton of wrappers for the various return types, and I don't think it's necessary for a first pass.

Instead, I want the RedisSource to return a C++ implementation of the interface, which we pass into the LazyLoad builder. 

This does mean the contract is entirely in documentation, there's no type safety here.